src/LinearA.hs

```diff
@@ -418,18 +447,20 @@ splitTangents scope env es = go scope env (freeVars <$> es)
       True  -> case commonFvs of
         [] -> (subcontext, subscope, (M.restrictKeys env fvs):submaps)
           where (subcontext, subscope, submaps) = go scope (M.withoutKeys env fvs) tfvs
-        _  -> (context . subcontext, subscope, (M.fromList $ zip commonFvs dvs2'):submaps)
+        _  -> (context . subcontext, subscope, this_map:submaps)
```

Comment:
completely minor nit: so far we've used camel case, so this is slightly inconsistent